ibm/service/contextbasedrestrictions/resource_ibm_cbr_zone.go

```diff
@@ -21,69 +20,70 @@ import (
 
 func ResourceIBMCbrZone() *schema.Resource {
 	return &schema.Resource{
-		CreateContext: resourceIBMCbrZoneCreate,
-		ReadContext:   resourceIBMCbrZoneRead,
-		UpdateContext: resourceIBMCbrZoneUpdate,
-		DeleteContext: resourceIBMCbrZoneDelete,
+		CreateContext: ResourceIBMCbrZoneCreate,
+		ReadContext:   ResourceIBMCbrZoneRead,
+		UpdateContext: ResourceIBMCbrZoneUpdate,
+		DeleteContext: ResourceIBMCbrZoneDelete,
 		Importer:      &schema.ResourceImporter{},
 
 		Schema: map[string]*schema.Schema{
-			"name": {
+			"name": &schema.Schema{
 				Type:         schema.TypeString,
```

Comment:
why some of required were changed to optional ...Are they really optional (in all resources i have seen it)

Reply:
made necessary changes